hash.go

```diff
@@ -18,8 +14,17 @@ func newDefaultHasher() Hasher {
 type fnv64a struct {
```

Comment:
Maybe we should move this structure and it's functions and constants to separate file and add tests for it. Also adding dockstring comments and pointing to original source is necessary.